src/lib/template-instance.ts

```diff
@@ -27,20 +27,20 @@ import {isTemplatePartActive, Template, TemplatePart} from './template.js';
  * with new values.
  */
 export class TemplateInstance {
-  private readonly __parts: Array<Part|undefined> = [];
```

Comment:
`Array<some complex type expression>` 

is more legible than 

`(some complex type expression)[]`